### PR TITLE
Give proper grace period when recorder is still in the room

### PR DESCRIPTION
**Pull request by davidzhao**
When a recorder is in the room, we would skip grace period due to a bug with how LastLeftAt was set. This would cause RoomComposite templates to exit immediately if the last participant in the room reconnects.

### CODE REVIEW

File: pkg/rtc/room.go [hunk 1]
```diff
@@ -464,6 +464,11 @@ func (r *Room) RemoveParticipant(identity livekit.ParticipantIdentity, pID livek
 	// send broadcast only if it's not already closed
 	sendUpdates := !p.IsDisconnected()
 
+	// remove all published tracks
+	for _, t := range p.GetPublishedTracks() {
```
Comment (davidzhao) [inline pkg/rtc/room.go:468]: drive-by. this cleans up tracks so any subscribers would get immediate removal.. instead of having SubscriptionManager keep trying to reconcile until timeout.

File: pkg/rtc/room.go [hunk 2]
```diff
@@ -597,16 +598,16 @@ func (r *Room) CloseIfEmpty() {
 		}
 	}
 
-	timeout := r.protoRoom.EmptyTimeout
+	var timeout uint32
 	var elapsed int64
-	if r.FirstJoinedAt() > 0 {
+	if r.FirstJoinedAt() > 0 && r.LastLeftAt() > 0 {
 		// exit 20s after
```
Comment (boks1971) [inline pkg/rtc/room.go:604]: can remove that specific `20s` comment or replace with a more generic comment.

**Review by cnderrauber — APPROVED** [review]

**Review by boks1971 — APPROVED** [review]
